app/models/concerns/package_quantity_validator.rb

```diff
@@ -0,0 +1,13 @@
+class PackageQuantityValidator < ActiveModel::Validator
+  def validate(record)
+    package = record.package
+    if(package.present? && is_invalid_total_quantity?(record))
+      record.errors.add(:quantity, "cannot be greater than package quantity")
+    end
+  end
+
+  def is_invalid_total_quantity?(record)
+    total = record.package.try(record.class.table_name).where.not(id: record.id).pluck(:quantity).sum + record.quantity
```

Comment:
I think this will work in that case because 
`record.package.try(record.class.table_name).where.not(id: record.id).pluck(:quantity).sum` is actually either of following two. 
```
record.package.try("packages_locations").where.not(id: record.id).pluck(:quantity).sum
or 
record.package.try("orders_packages").where.not(id: record.id).pluck(:quantity).sum
``` 
So if its a new record and `record.id` is nil then we'll get `record.package.try("orders_packages").where.not(id: record.id) = [ ]` and if we use `pluck(:quantity).sum` on this we get answer as 0. 
Still to be sure i tried following on my local db:
```
2.2.2 :009 > Package.first.packages_locations
 => #<ActiveRecord::Associations::CollectionProxy []>

2.2.2 :011 > Package.first.packages_locations.where.not(id: nil).pluck(:quantity).sum
 => 0 
```

So by the above check I think its safe as we trying to search for "orders_package" or "packages_location" on the parent package and if there isn't any record, `pluck` will return nil and `sum` will turn that into 0. So I think its safe. Do let me know if there is something i am missing.